dashboard/models.py

```diff
@@ -172,17 +172,22 @@ def get_data_last_updated(self) -> Optional[datetime]:
         """ Returns the datetime of the last cron run of all courses
             This checks for any courses where the data_last_updated value is null.
 
+            If there are new courses it will return the lower value of the last run date or the earliest start date.
+
         :return: datetime. Either the earliest or None
-        :rtype: None
+        :rtype: None, datetime
         """
         new_courses = self.filter(data_last_updated__isnull=True)
+        existing_courses = self.filter(data_last_updated__isnull=False)
 
+        existing_earliest = existing_courses.earliest("data_last_updated").data_last_updated
         # If there are new courses (courses with no last run) return the earliest time of all
+        new_earliest = None
         if len(new_courses) > 0:
-            return new_courses.earliest_start_datetime()
-        # Otherwise return the latest cron run
-        else:
-            return self.all().latest("data_last_updated").data_last_updated
+            new_earliest = new_courses.earliest_start_datetime()
+
+        # Return the lower value of existing_earliest and new_earliest, otherwise return None
+        return min((val for val in [existing_earliest, new_earliest] if val is not None), default=None)
```

Comment:
I'm finding this difficult to reason about so I'm writing some basic pseudocode describing the behavior I think we should be aiming for (not necessarily just in these methods, but in general in determining the cutoff for the cron's queries).

```
if not new_courses:
   return data_last_updated
else:
   if not data_last_updated (all courses are new) or new_courses.earliest_start_datetime < data_last_updated:
      return new_courses.earliest_start_datetime
   else:
      return data_last_updated
```

Aren't we kind of missing the cases where we would still want to use the `data_last_updated` (when all courses are old or all the new courses are more recent than when the data was last updated)? Still seems like we'd want to avoid fetching data we already have. I'm also not sure we need to know the `existing_earliest`?

Reply:
Okay, wait, I was misreading what was going on with `existing_earliest`. Let me re-examine.

Reply:
Yeah, what you have is basically what I wrote in pseudocode, maybe even more concise. I think `existing_earliest` could be better named, since the naming of `new_earliest` made me keep thinking it was using `earliest_start_datetime`. I think I'd prefer just `data_last_updated`.

Reply:
Thanks for thinking about it. I did basically the same thing and tested it in a few scenarios by manipulating the database. I think the test plan on the issue should be pretty good and is what is expected by this pseudocode. 

I basically do the same thing by returning the min value of the two rather than doing some other date checks here. Really I just want. I can look at the variable names and try to improve them.

Really what I was thinking of is that last line

# New courses means that the data has never been updated
# Data Last Updated is a field added once courses have been processed at least once
`return min(earliest_data_last_updated, earliest_start_date_of_new_courses)
`
However `earliest_data_last_updated` would be `None` on the first run. `earliest_start_date_of_new_courses` would be `None` if there are no new courses. Both would be `None` if there are no courses at all and cron is run. 

If one of them is `None` this caused a problem so had to protect this with a for loop. And if **both** of them are `None`, still should just return `None` since cron is protecting this case. This could have been a lot more if statements as your pseudocode has, I just felt like I'd miss a check.

Reply:
Renamed these methods to match the example here